api/models/Activity.js

```diff
@@ -101,13 +136,86 @@ module.exports = bookshelf.Model.extend({
       reader_id: post.get('user_id'),
       actor_id: unfollowerId,
       post_id: post.id,
-      action: this.Action.Unfollow,
+      meta: {reasons: [this.Reason.Unfollow]},
       created_at: new Date()
     })
   },
 
   unreadCountForUser: function (user) {
     return Activity.query().where({reader_id: user.id, unread: true}).count()
     .then(rows => rows[0].count)
+  },
+
+  mergeReasons: function (reasons) {
+    const merged = values(reasons.reduce((acc, reason) => {
```

Comment:
actually i would maybe just pull this out into a function at the top level of the file since it has a pretty specific purpose and is unlikely to ever be used outside the file